packages/protocol/contracts/governance/Election.sol

```diff
@@ -475,14 +501,55 @@ contract Election is
    * @return The total number of votes cast by an account.
    */
   function getTotalVotesByAccount(address account) external view returns (uint256) {
-    uint256 total = 0;
+    bool allowedToVoteOverMaxNumberOfGroupsForAccount = allowedToVoteOverMaxNumberOfGroups[account];
     address[] memory groups = votes.groupsVotedFor[account];
+    require(
+      allowedToVoteOverMaxNumberOfGroupsForAccount || groups.length < maxNumGroupsVotedFor,
+      "Too many groups voted for"
+    );
+
+    if (groups.length > maxNumGroupsVotedFor) {
+      uint256 totalVotes = cachedVotesByAccount[account].totalVotes;
+      require(totalVotes > 0, "Total votes are not counted! Use updateTotalVotesByAccount.");
+      return totalVotes;
+    }
+
+    uint256 total = 0;
     for (uint256 i = 0; i < groups.length; i = i.add(1)) {
       total = total.add(getTotalVotesForGroupByAccount(groups[i], account));
     }
     return total;
   }
 
+  /**
+   * @notice Counts and caches account's votes for group.
+   * @param account The address of the voting account.
+   * @param group The address of the validator group.
+   */
+  function updateTotalVotesByAccountForGroup(address account, address group) external {
+    cachedVotesByAccount[account].cachedVotesPerGroup[group] = getTotalVotesForGroupByAccount(
+      group,
+      account
+    );
+  }
+
+  /**
+   * @notice Counts and caches account's total votes.
+   * @param account The address of the voting account.
+   */
+  function updateTotalVotesByAccount(address account) external {
```

Comment:
I would not have this function - it can still run into that out of gas problem we're trying to avoid if there's too many groups. Instead, in `updateTotalVotesByAccountForGroup`, update the total incrementally, i.e. subtract the previous value, add the new value to the total.

Reply:
When the above change is made, I think `updateTotalVotesByAccountForGroup` could just be used in all the other places we update the total - e.g. when voting, unvoting.